flow/src/lib.rs

```diff
@@ -836,6 +734,182 @@ impl<T: Config> Pallet<T> {
 		Nonce::<T>::put(nonce.wrapping_add(1));
 		nonce
 	}
+
+	fn schedule_campaign_settlements(block_number: T::BlockNumber) {
+		// which campaigns end in this block
+		let campaign_ids = CampaignsByBlock::<T>::get(block_number);
+
+		// iterate over campaigns ending in this block
+		for campaign_id in &campaign_ids {
+			let campaign = Campaigns::<T>::get(campaign_id);
+			let campaign_balance = CampaignBalance::<T>::get(campaign_id);
+			// check for cap reached
+			if campaign_balance >= campaign.cap {
+				Self::set_state(campaign.id, FlowState::Finalizing);
+
+				Self::deposit_event(Event::CampaignFinalising {
+					campaign_id: *campaign_id,
+					campaign_balance,
+					block_number,
+					success: false,
+				});
+
+			// campaign cap not reached
+			} else {
+				Self::set_state(campaign.id, FlowState::Reverting);
+
+				Self::deposit_event(Event::CampaignReverting {
+					campaign_id: *campaign_id,
+					campaign_balance,
+					block_number,
+					success: false,
+				});
+			}
+		}
+	}
+
+	fn process_campaigns(block_number: &T::BlockNumber, state: FlowState, processed: &mut u32) -> Weight {
+		let campaign_ids = CampaignsByState::<T>::get(&state);
+		let total_weight: Weight = 0;
+		for campaign_id in campaign_ids {
+			let campaign = Campaigns::<T>::get(campaign_id);
+			let campaign_balance = CampaignBalance::<T>::get(campaign_id);
+			let org = CampaignOrg::<T>::get(&campaign_id);
+			let org_treasury = T::Control::org_treasury_account(&org);
+			let contributors = CampaignContributors::<T>::get(campaign_id);
+
+			if state == FlowState::Finalizing {
+				if let Some(owner) = CampaignOwner::<T>::get(campaign.id) {
+					total_weight.saturating_add(
+						Self::finalize_campaign(&block_number, processed, &campaign, &campaign_balance, &org, &org_treasury, &contributors, &owner)
+					);
+				} else {
+					// TODO: If no campaign owner, what should be done?
```

Comment:
- wasn't this a check for if a campaign contribution came from an owner, not to take it? 
- in general a campaign must not have no owner and should always have the dao as owner

Reply:
It was a check if `CampaignOwner` has a value.
[https://github.com/gamedaoco/gamedao-protocol/blob/main/flow/src/lib.rs#L395](https://github.com/gamedaoco/gamedao-protocol/blob/main/flow/src/lib.rs#L395)

Reply:
ah right, just a safety check, so the funds are not sent into nirvana. I thought you meant this here: https://github.com/gamedaoco/gamedao-protocol/blob/39585b0036ec05a6078bda2ff99e901579b0f579/flow/src/lib.rs#L405

Reply:
to answer your question, if sth bad like this happens, it should fail gracefully with an event noting a missing treasury